### ISSUE TITLE
cloudwatch: Metric created x times result in different dimensions

### ISSUE THREAD

**Issue by kornicameister** [aws/aws-cdk#21478]
### Describe the bug

I have a stack that involves two construct. One sets up the dashboard using set of custom metrics and the other one sets up alarms for the very same metrics. Essentially both construct have, inside their code, a call like this:

```
                    cw.Metric(
                        label=name,
                        namespace='CWAgent',
                        metric_name='collectd_processes_value',
                        dimensions_map={
                            'instance': name,
                            'InstanceId': instance_id,
                            'type': 'ps_count',
                        },
                        unit=cw.Unit.COUNT,
                        statistic='max',
                    ),
```

As per documentation I think this is fine because `Metric` is not a construct, not to mention that even if they were constructs they are created in separate scope so there wouldn't be conflict. Anyway, the metric that is generated for dashboard looks and works perfectly fine because CloudWatch renders the dashboard correctly. The one that is used to create alarms i.e. `metric.createAlarm` is broken and the only real difference I was able to spot is the order of dimensions in resulting AWS Alarm resource.

The one that works has: `instance, InstanceId, type` (dashboard)
The one that does not work has: `instance, type, InstanceId` (alarm)

So, in other words, the one that works has the dimensions presented in the same order as in the code.

### Expected Behavior

Regardless how many times `Metric` is created it always has dimensions ordered as presented in the code.

### Current Behavior

Alarm from metric 2nd time does not work.

### Reproduction Steps

TBD but I assume it is enough to have a stack that creates metric two times and creates two distinct alarms out of it.

### Possible Solution

N/A

### Additional Information/Context

Python 3.10

### CDK CLI Version

2.33.0

### Node.js Version

v14.17.5

### OS

MacOs BigSur

### Language

Python

### Language Version

3.10.1

**Comment by kornicameister** [comment]
I have tried removing the alarms, redeploying and bringing them back up but that didn't really make any difference.
If I navigate to edit page of an alarm I see dimensions out of order and no datapoint form alarm. If I try to create alarm manually and select a metric dimensions there are ordered as expected thus there are datapoints.

**Comment by kornicameister** [comment]
But, if someone can explain to me why that was an issue...I'd be grateful.
Perhaps that's obvious but best to ask.
